### ISSUE TITLE
client-s3 :: Setting `AccessControlPolicy` via `putObjectAcl` throws a MalformedACLError

### ISSUE THREAD

**Issue by philiiiiiipp** [aws/aws-sdk-js-v3#2267]
## Describe the bug

I tried setting an AccessControlPolicy using the `putObjectAcl` but it seems to throw a `MalformedACLError`

```
Error [MalformedACLError]: The XML you provided was not well-formed or did not validate against our published schema
    at /var/task/functions/resolver/webpack:/Users/philipp/node/dev/dh/sfa-api/node_modules/@aws-sdk/client-s3/dist/es/protocols/Aws_restXml.js:8733:1
    at step (/var/task/functions/resolver/webpack:/Users/philipp/node/dev/dh/sfa-api/node_modules/@aws-sdk/client-s3/node_modules/tslib/tslib.es6.js:102:1)
    at Object.next (/var/task/functions/resolver/webpack:/Users/philipp/node/dev/dh/sfa-api/node_modules/@aws-sdk/client-s3/node_modules/tslib/tslib.es6.js:83:1)
    at fulfilled (/var/task/functions/resolver/webpack:/Users/philipp/node/dev/dh/sfa-api/node_modules/@aws-sdk/client-s3/node_modules/tslib/tslib.es6.js:73:1)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  Code: 'MalformedACLError',
  RequestId: 'XXX',
  HostId: 'XXX,
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    requestId: undefined,
    extendedRequestId: 'XXX',
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  }
} 
```

My request is

```
putObjectAcl({
    Bucket: 'somebucket',
    Key: 'somekey',
    AccessControlPolicy: {
      Grants: [
        {
          Grantee: {
            DisplayName: 'somename',
            ID: 'someid',
            Type: 'CanonicalUser',
          },
          Permission: 'FULL_CONTROL',
        },
      ],
      Owner: {
            DisplayName: 'somename',
            ID: 'someid',
      },
    },
  });
```

When trying the same request but setting an ACL instead, the request works just fine.

```
putObjectAcl({
    ACL: 'public-read',
    Bucket: 'somebucket',
    Key: 'somekey',
  });
```

#### SDK version number

```
├── @aws-sdk/client-s3@3.12.0 
└─┬ @aws-sdk/lib-storage@3.12.0
  └── @aws-sdk/client-s3@3.12.0 
```

#### Is the issue in the browser/Node.js/ReactNative?
Node.js

#### Details of the browser/Node.js/ReactNative version
It runs inside a lambda function set to `nodejs12.x`

**Comment by ajredniwja** [comment]
HI @philiiiiiipp thanks for opening this issue, I was able to reproduce this, both in V2 and V3 with root cause still unknown, I have reached out to S3 team to have a look.

**Comment by kgwong-uc** [comment]
This seem to be fixed since `3.18.0`
